src/components/Toggle.tsx

```diff
@@ -1,139 +1,121 @@
+import { CheckDisable, Change } from './common-types'
+import { ifStyle } from '../lib/if-prop'
 
-interface ToggleProps {
-  disabled?: boolean;
-  checked?: boolean;
-  onChange?: (event: boolean) => void;
-}
+type ToggleProps = Partial<CheckDisable> & Change
 
-const Label: FC<{ disabled: boolean }> = ({ disabled, ...rest }) => (
-  <label
+const Handler: FC<CheckDisable> = ({ checked, disabled }) => (
+  <span
     sx={{
-      // '-webkit-tap-highlight-color': '0',
-      display: "inline-block",
-      verticalAlign: "middle",
-      whiteSpace: "nowrap",
-      userSelect: "none",
-      py: 1,
-      px: 0,
-      position: "relative",
-      cursor: disabled ? "not-allowed" : "pointer",
-      opacity: disabled ? 0.4 : 1,
+      width: 'calc(0.875rem - 3px)',
+      height: 'calc(0.875rem - 3px)',
+      position: 'absolute',
+      top: '50%',
+      transform: 'translateY(-50%)',
+      left: '1px',
+      boxShadow: 'rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.2) 0 1px 3px 0',
+      transition: 'left 280ms cubic-bezier(0, 0, 0.2, 1)',
+      borderRadius: '50%',
+      border: '1px solid transparent',
+      backgroundColor: 'background',
+      ...ifStyle(checked, {
+        left: 'calc(100% - (0.875rem - 2px))',
+        boxShadow: 'none',
+      }),
+      ...ifStyle(disabled, {
+        backgroundColor: 'darkGray',
+        boxShadow: 'none',
+        borderColor: 'darkGray',
+      }),
     }}
-    {...rest}
   />
-);
+)
 
-type OnInputChange = (event: ChangeEvent<HTMLInputElement>) => void;
-const Input: FC<{
-  type: string;
-  disabled: boolean;
-  checked: boolean;
-  onChange: OnInputChange;
-}> = (props) => (
-  <input
-    sx={{
-      overflow: "hidden",
-      visibility: "hidden",
-      height: 0,
-      opacity: 0,
-      width: 0,
-      position: "absolute",
-      backgroundColor: "transparent",
-      zIndex: -1,
-    }}
-    {...props}
-  />
-);
-const ToggleHandler: FC<{
-  disabled: boolean;
-  checked: boolean;
-}> = ({ disabled, checked, ...rest }) => (
-  <div
-    sx={{
-      width: "1.75rem",
-      height: "0.875rem",
-      borderRadius: "0.875rem",
-      transitionDelay: "0.12s",
-      transitionDuration: "0.2s",
-      transitionProperty: "background, border",
-      transitionTimingFunction: "cubic-bezier(0, 0, 0.2, 1)",
-      position: "relative",
-      border: "1px solid transparent",
-      backgroundColor: "lightGray",
-      padding: 0,
-      ...(checked
-        ? {
-            backgroundColor: "primary",
-          }
-        : {}),
-      ...(disabled
-        ? {
-            borderColor: "darkGray",
-          }
-        : {}),
-
-      ...(disabled && checked
-        ? {
-            backgroundColor: "lightGray",
-          }
-        : {}),
-    }}
-    {...rest}
-  >
-    <span
+export const Body: FC<
+  CheckDisable & { onChange: (e: ChangeEvent<HTMLInputElement>) => void }
+> = ({ checked, disabled, children, onChange }) => (
+  <Fragment>
```

Comment:
Why using Fragment instead of `<>`?

Reply:
With `<>` I get the error `JSX fragment is not supported when using an inline JSX factory pragmats(17017)`.

Do you have it?
Not really sure what does it means... any ideas?

Reply:
is going to be some config